### ISSUE TITLE
Does ignite support to load weights for different num_classes?

### ISSUE THREAD

**Issue by Nic-Ma** [pytorch/ignite#811]
## ❓ Questions/Help/Support
Hi @vfdev-5 ,

Does ignite Checkpoint handler support the case that pre-trained model has different number output with the model load to?
For example, the weights in model checkpoint is for 1000 classes output, and now user defines the network with 3 classes output and use ignite handler to load pretrained model for the weights.
Thanks.

**Comment by justusschock** [comment]
Hi @Nic-Ma 
I think this is currently not supported, because we simply load the checkpoint and don't do any checks on the Tensor sizes. 
We should allow kwargs in the `load_objects` function and you would have to pass everything except the last layer as `to_load`. I think implementing a general interface for that is not that easy, but I'll think on that.

**Comment by vfdev-5** [comment]
Hi @Nic-Ma ,

yes, as @justusschock said this is not supported and probably the reason for that is related to the fact that pretrained weights loading in most of the cases provided by model definition (e.g. in torchvision `resnet18(pretrained=True)`). In that case, user's modification can be easily done a posteriori :
```python
model = resnet18(pretrained=True)
model.fc = nn.Linear(512, 3)
```

**Comment by justusschock** [comment]
@vfdev-5 we should nevertheless think about kwargs for the load_objects function, to allow kwargs for both, `torch.load` (for example the `map_location`) and the `load_state_dict` (for example the `strict`).

**Comment by vfdev-5** [comment]
Yes, I agree!

**Comment by Nic-Ma** [comment]
Hi @vfdev-5 and @justusschock ,

Thanks very much for your sharing. Looking forward to this feature.
It's really useful for us, especially to provide "Transfer Learning" function in our product.
Currently, we can't exclude FC layer when loading weights, users raised this request.
Thanks.

**Comment by vfdev-5** [comment]
Closed by #861
